### PR TITLE
Port core storage to T1

### PR DESCRIPTION
**Pull request by hiviah**
Port pythonic storage from Core to work on T1 (#1015).

Includes cache and resident credentials, maximum count of resident credentials is limited to 20 for now due to smaller flash space for storage.

### CODE REVIEW

File: core/embed/trezorhal/flash.c [hunk 1]
```diff
@@ -52,6 +53,9 @@ static const uint32_t FLASH_SECTOR_TABLE[FLASH_SECTOR_COUNT + 1] = {
     [22] = 0x081C0000,  // - 0x081DFFFF | 128 KiB
     [23] = 0x081E0000,  // - 0x081FFFFF | 128 KiB
     [24] = 0x08200000,  // last element - not a valid sector
+#elif TREZOR_MODEL == 1
+    [12] = 0x08100000,  // - last element - not a valid sector
```
Comment (prusnak) [inline core/embed/trezorhal/flash.c:57]:
```suggestion
    [12] = 0x08100000,  // last element - not a valid sector
```

File: core/embed/unix/flash.c [hunk 2]
```diff
@@ -60,6 +61,9 @@ static const uint32_t FLASH_SECTOR_TABLE[FLASH_SECTOR_COUNT + 1] = {
     [22] = 0x081C0000,  // - 0x081DFFFF | 128 KiB
     [23] = 0x081E0000,  // - 0x081FFFFF | 128 KiB
     [24] = 0x08200000,  // last element - not a valid sector
+#elif TREZOR_MODEL == 1
+    [12] = 0x08100000,  // - last element - not a valid sector
```
Comment (prusnak) [inline core/embed/unix/flash.c:65]:
```suggestion
    [12] = 0x08100000,  // last element - not a valid sector
```

**Comment by prusnak** [comment]
@hiviah I pushed a fixup commit which adds `trezor/messages/BackupType.py` so we can reuse `core/src/storage/device.py` as a symlink. Can you please check if this still works as expected?

**Comment by hiviah** [comment]
Yes, it works still as expected (both on device and emulator).

**Comment by prusnak** [comment]
Thanks!

**Pull request merged**